src/Microsoft.IdentityModel.Tokens/LogMessages.cs

```diff
@@ -89,7 +89,7 @@ internal static class LogMessages
 
         // 10500 - SignatureValidation
         public const string IDX10500 = "IDX10500: Signature validation failed. No security keys were provided to validate the signature.";
-        public const string IDX10501 = "IDX10501: Signature validation failed. Unable to match keys: \nkid: '{0}', \ntoken: '{1}'.";
+        public const string IDX10501 = "IDX10501: Signature validation failed. Unable to match keys: \nkid: '{0}'.\nExceptions caught:\n '{1}'. \ntoken: '{2}'.";
```

Comment:
nit: (was there before) Could we change - Unable to match keys: -> Unable to match key:  (singular).

Reply:
That's a good idea! Makes the message a lot clearer. 

Just made this fix.

Reply:
thanks! :shipit: